src/OrleansProviders/Streams/Generator/GeneratorPooledCache.cs

```diff
@@ -58,13 +60,65 @@ public bool Equals(CachedMessage cachedMessage, IStreamIdentity streamIdentity)
             }
         }
 
+        private class ExplicitEvictionStrategy : IEvictionStrategy<CachedMessage>
```

Comment:
This looks like the same logic as ExplicitEvictionStrategy in memory provider.  If so we should make common, not duplicate.